Upstream-Status: Pending

If /etc/mtab is a symlink to /proc/self/mounts,
mounting fails because addmntent returns an error.
As a consequence, the Private directory is unusable.
This is a patch which checks if /etc/mtab is a symlink,
and if so, doesn't try to call addmntent.

Signed-off-by: Guo Chunrong <B40290@freescale.com>


--- a/hugeadm.c	2013-07-31 23:46:04.461258351 -0500
+++ b/hugeadm.c	2013-07-31 23:50:03.249359534 -0500
@@ -552,6 +552,16 @@
 			return 1;
 		}
 
+         char dummy;
+         int useMtab;
+          /* Check if mtab is a symlink */
+          useMtab = (readlink(MOUNTED, &dummy, 1) < 0);
+        if (!useMtab) {
+         /* No need updating mtab */
+          return 0;
+        }
+  
+
 		mounts = setmntent(MOUNTED, "a+");
 		if (mounts) {
 			entry.mnt_fsname = FS_NAME;
